### PR TITLE
build: update go 1.16 to 1.17

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
@iamniting is it tested for PVC operations or need to be tested?

File: go.mod [hunk 1]
```diff
@@ -20,3 +20,78 @@ require (
 	sigs.k8s.io/controller-runtime v0.10.2
 	sigs.k8s.io/yaml v1.2.0
 )
+
+require (
```
Comment (leelavg) [inline go.mod:24]: - any idea why this was factored out into a new `require` block and no changes to `go.sum`?

Comment (iamniting) [reply go.mod:24]: Actually, I ran `go mod tidy` which did this change, and nothing in the go.mod file. As they are all indirect deps we might already have all of them in the go.sum

**Comment by iamniting** [comment]
> @iamniting is it tested for PVC operations or need to be tested?

Not yet

**Comment by iamniting** [comment]
> > @iamniting is it tested for PVC operations or need to be tested?
> 
> Not yet

Tested it out and it works well

```
┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  golang]
└──╼$ oc get pods
NAME                                               READY   STATUS    RESTARTS   AGE
app-file                                           1/1     Running   0          27s
lvm-operator-controller-manager-547b5bbb46-ml9d4   3/3     Running   0          4m5s
topolvm-controller-778cfb5d54-8xfln                4/4     Running   0          3m5s
topolvm-node-bjhtk                                 4/4     Running   0          3m5s
vg-manager-6ffc2                                   1/1     Running   0          3m5s

┌─[nigoyalॐ nigoyal ~/code/go/src/github.com/red-hat-storage/lvm-operator  golang]
└──╼$ oc get pvc
NAME           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
lvm-file-pvc   Bound    pvc-39711d19-4f15-448c-ab0e-9a28c3e58732   5Gi        RWO            odf-lvm-vg1    65s


```

**Review by leelavg — APPROVED** [review]
/lgtm

**Review by nbalacha — APPROVED** [review]

**Pull request merged**